### DISCUSSION TITLE
Autoattachment

### DISCUSSION THREAD

**Discussion by Keanottesen** [xyflow/xyflow#1748] [Ideas]
Is it possible to create an auto attachment. So if our connectionline is with x proximity, it tries to attach to the handle with a magnetism effect?

**Comment by carlosjedwab** [comment]
@Keanottesen Hey, I had a similay problem where the distance between the nodes was an important factor as well. It was not the same exact problem, but I needed to **autoallign** the nodes at the center, to make it easier to position them side by side. Here is the code: https://codesandbox.io/s/react-flow-autoallign-pvm32?file=/src/App.js

I bet you can change the logic I used for the `onNodeDrag` to, instead of changing the snap grid, force the connection between nodes with a `onConnect.`

I'm sorry if the code is a little confusing to read, but I hope it helps.

**Reply by carlosjedwab** [reply]
The logic is simple. I check if the nodes are align with each other. If so, I increase the snap grid to make it stay in the axis position for a little longer

**Reply by Keanottesen** [reply]
Thanks for the example. But I fail to see how I can implement it in my situation. I do not in a custom connectionline know the position of all other nodes and the respective handles, thus I think I am not able to implement such a logic.

**Reply by carlosjedwab** [reply]
So I made this other code: https://codesandbox.io/s/react-flow-autoatach-p6mw3?file=/src/App.js

And in fact, I had to assume the handle is vertical to the node, and in the middle of the node

**Comment by datoslabs** [comment]
I suppose you can implement custom nodes and handles and affect the handle's CSS to a larger size (e.g. increase the border size or add shadow) via `onConnect` handler.  I have, in the past, changed the handle's SVG to make it easier for users to connect; however, I resorted to implementing my own point-n-click style connection mechanism.

**Reply by Keanottesen** [reply]
That is a clever idea. I might try implementing such a solution.
